spec/features/users/brouillon_spec.rb

```diff
@@ -196,25 +203,14 @@
     expect(page).to have_text('RIB.pdf')
   end
 
-  # TODO: once we're running on Rails 6, re-enable the validator on PieceJustificativeChamp,
-  # and unmark this spec as pending.
-  #
-  # See piece_justificative_champ.rb
-  # See https://github.com/betagouv/demarches-simplifiees.fr/issues/4926
-  scenario 'add an invalid attachment', js: true, pending: true do
-    log_in(user, procedure_with_pjs)
+  scenario 'add an invalid attachment on an old procedure where pj validation is disabled', js: true do
+    log_in(user, old_procedure_with_disabled_pj_validation)
     fill_individual
 
     # Test invalid file type
     attach_file('Pièce justificative 1', Rails.root + 'spec/fixtures/files/invalid_file_format.json')
```

Comment:
```suggestion
    attach_file('Pièce justificative 1', Rails.root + 'spec/fixtures/files/valid_file_format.json')
```

Reply:
est ce qu'il ne faudrait pas simuler une grosse taille de pj ?

Reply:
Ce serait bien mais je n'arrive pas à écrire ce test.
 - on ne peut pas passer par un faux fichier, car capybara veut vraiment faire l'upload d'un fichier (ce serait top de pouvoir faire `allow(attestation).to receive(:pdf).and_return(double(read: 'pdf', size: 2.megabytes))`)
 - j'ai pensé générer un gros fichier temporaire pour l'utiliser dans mon test

```
large_tmp_file = Tempfile.new('large_file')
large_tmp_file.write('s' * 210.megabytes)
large_tmp_file.close
```

*mais* c'est pas fou, et surtout **Capybara semble en PLS avec les gros fichiers** : je n'arrive pas à faire uploader un fichier lourd dans un test.

 - avec un fichier léger, ce test passe. Je modifie le timeout exprès pour que `wait_until` aboutisse, même si ce n'est pas nécessaire pour ce premier test.

```ruby
  scenario 'add an invalid attachment on procedure with validation', js: true do
    log_in(user, procedure_with_pj)
    fill_individual

    Capybara.default_max_wait_time = 30
    # avec un fichier très léger, ça passe
    attach_file('Pièce justificative', '/home/.../tps/README.md')   
    blur
    wait_until { user.reload.dossiers.last.champs.first.piece_justificative_file.attached? == true }

    expect(user.reload.dossiers.last.champs.first.piece_justificative_file.attached?).to be_truthy
  end
```
 - si dans `attach_file`, je mets un gros fichier (300 Mo), même avec 60s de timeout je finis par avoir un `Timeout::Error: execution expired`. Le timeout est bien pris en compte, l'exécution du test prend plus d'une minute

```ruby
  attach_file('Pièce justificative', '/home/.../fichier-tres-lourd')
```

L'appli rails n'est pas très véloce, même à la main en local; je ne trouve rien à ce sujet sur le net
Si quelqu'un a une idée, je prends, sinon, on peut se dire qu'on a des tests au niveau du modèle.

Reply:
Solution possible: mettre la taille max en variable d'env pour le surcharger dans l'environnement de test. Ca parait un changement un peu pénible pour un test déja présent coté modèle, avec de toute façon un environnement différent de la prod, pour tester une fonctionnalité d'une gem. On a déja des tests sur l'affichage des erreurs de validation ailleurs, donc j'ai l'impression qu'on peut se passer de celui là.